ompi/mca/osc/sm/osc_sm_component.c

```diff
@@ -291,7 +292,10 @@ component_select(struct ompi_win_t *win, void **base, size_t size, int disp_unit
                                                   rbuf, 1, MPI_UNSIGNED_LONG,
                                                   module->comm,
                                                   module->comm->c_coll->coll_allgather_module);
-        if (OMPI_SUCCESS != ret) return ret;
+        if (OMPI_SUCCESS != ret) {
+            free(rbuf);
+            return ret;
```

Comment:
Why return here and not go-to the error state? Looks like that might potentially lead to another leak?

Reply:
The original code was just a **#return ret** but it looks like that wasn't right either. There's error flows before and after this that jump to cleanup. There's also an error path around line 315 that needs to be fixed.

So I need to fix this in main, where this chagne has been merged, then cherry pick that commit to v5